docs/pages/api/index.md

```diff
@@ -12,33 +12,115 @@ env = gym.make('CartPole-v0')
+
+The agent performs some actions in the environment (usually by passing some control inputs to the environment, e.g. torque inputs of motors) and observes
+how the environment's state changes. One such action-observation exchange is referred to as a *timestep*. 
+
+The goal in RL is to manipulate the environment in some specific way. For instance, we want the agent to navigate a robot
+to a specific point in space. If it succeeds in doing this (or makes some progress towards that goal), it will receive a positive reward
+alongside the observation for this timestep. The reward may also be negative or 0, if the agent did not yet succeed (or did not make any progress). 
+The agent will then be trained to maximize the reward it accumulates over many timesteps.
+
+After some timesteps, the environment may enter a terminal state. For instance, the robot may have crashed! In that case,
+we want to reset the environment to a new initial state. The environment issues a done signal to the agent if it enters such a terminal state.
+Not all done signals must be triggered by a "catastrophic failure": Sometimes we also want to issue a done signal after
+a fixed number of timesteps, or if the agent has succeeded in completing some task in the environment.
+
+Let's see what the agent-environment loop looks like in Gym.
+This example will run an instance of `LunarLander-v2` environment for 1000 timesteps, rendering the environment at each step. You should see a window pop up rendering the environment
 
 ```python
 import gym
-env = gym.make('CartPole-v0')
-env.reset()
-for _ in range(1000): 
-	env.render()  # by default `mode="human"`(GUI), you can pass `mode="rgb_array"` to retrieve an image instead
-	env.step(env.action_space.sample())  # take a random action 	
+env = gym.make("LunarLander-v2")
+env.action_space.seed(42)
+
+observation, info = env.reset(seed=42, return_info=True)
+
+for _ in range(1000):
+    env.render()
+    observation, reward, done, info = env.step(env.action_space.sample())
+
+    if done:
+      observation, info = env.reset(return_info=True)
+
 env.close()
 ```
 
 The output should look something like this
 
-![cartpole-no-reset](https://user-images.githubusercontent.com/28860173/129241283-70069f7c-453d-4670-a226-854203bd8a1b.gif)
+![lunar_lander](https://user-images.githubusercontent.com/15806078/153222406-af5ce6f0-4696-4a24-a683-46ad4939170c.gif)
+
+
+## Commonly used methods
+>### Stepping
+> `step(action: ActType) -> Tuple[ObsType, float, bool, dict]`
+> 
+>Run one timestep of the environment's dynamics. When end of episode is reached, you are responsible for calling `reset()`
+to reset this environment's state. Accepts an action and returns a tuple `(observation, reward, done, info)`.
+>
+>>**Parameters**
+>>- `action`(**object**): an action provided by the agent. This will oftentimes be a numpy array, but might also be 
```

Comment:
Maybe mention that an action is often described by a `gym.Space`? Same for observations.

Actually, it might be good to mention action and observation spaces before the methods (now it's at the end), -- and that observation spaces and action spaces are described by a Space, which is typically either a Box or a Discrete (or sometimes a Dict) and so on.

Reply:
I am now briefly mentioning `observation_space` and `action_space` after the minimal example. They are then discussed in more detail in the section "Additional Environment API", followed by a section on Spaces. Do you think that's sufficient?
I didn't want to get lost in all the intricate details on spaces at the very beginning of the page so I pushed some of it further to the back.